### ISSUE TITLE
EditorExportPlugin: Option to avoid exporting files

### ISSUE THREAD

**Issue by neikeq** [godotengine/godot#4054]
Currently, the method `Variant custom_export(String name, EditorExportPlatform platform)` in EditorExportPlugin allows to change the raw contents and the name of the file being exported. This is the method description:

```
1) If returned value is null, the file is exported
as is.

2) If the returned value is a RawAray, the content of that array becomes the new
file being exported.

3) If the file must also change it's name when
exported, then a [Dictionary] must be returned with
two fields: 'name' with the new filename and 'data'
with a [RawArray] containing the raw contents of the
file.
```

This method should also allow not to export the file, e.g:

```
4) If the file must not be exported, then an empty [Dictionary] must be returned.
```

**Comment by kubecz3k** [comment]
+1 from me, it would be nice to have an option to NOT export file what's only needed by plugin during the editor time.

**Comment by neikeq** [comment]
This is also required for the new plugins API. Plugins are project-specific now, and many plugins may contain files that are only used in the editor. There must be a way from the plugin context to specify that those files must not be exported.
I am not sure my example solution would be the best for plugins though. It may be better to just have a property of type array in `plugins.cfg`:

``` ini
export_ignore = [ "dont_export_me_plz.gd", "editor_image.png" ]
```

My example for `Variant custom_export(String name, EditorExportPlatform platform)` is still useful in other situations.

**Comment by neikeq** [comment]
Looks like this was fixed by d3c1f2a :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada:

**Comment by neikeq** [comment]
Now implementing the `export_ignore` option for `plugins.cfg` should be possible!
